### PR TITLE
[DPE-5767] - Add CI workflows

### PR DESCRIPTION
**Pull request by skourta**
I rebased the CI PR and opened it against the build branch.
This pull request includes several changes to the repository, focusing on configuration updates, workflow enhancements, and documentation improvements. The most important changes are listed below:

### Workflow Enhancements:
* Introduced a new GitHub Actions workflow in `.github/workflows/ci.yaml` for building and testing the snap package, including steps for downloading, installing, and running tests on the snap package.
* Added a configuration file for syncing GitHub issues to Jira in `.github/workflows/.jira_sync_config.yaml`, specifying settings like components, label mapping, and status mapping.

### Documentation Improvements:
* Created a new bug report issue template in `.github/workflows/ISSUE_TEMPLATE/bug_report.md` to standardize bug reporting with sections for steps to reproduce, expected behavior, actual behavior, versions, log output, and additional context.

### CODE REVIEW

File: .github/workflows/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Juju CLI: 
+
+<!-- Model version from `juju status` -->
+Juju agent: 
+
+<!-- App revision from `juju status` or (advanced) commit hash -->
+Charm revision: 
+
+<!-- Run `lxd version` -->
+LXD:
```
Comment (reneradoi) [inline .github/workflows/ISSUE_TEMPLATE/bug_report.md]: Not required here, maybe `snap revision` instead.

File: .github/workflows/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+<!-- Run `lxd version` -->
+LXD: 
+
+## Log output
+<!-- Please enable debug logging by running `juju model-config logging-config="<root>=INFO;unit=DEBUG"` (if possible) -->
+<!-- Then, run `juju debug-log --replay > log.txt` and upload "log.txt" file here -->
+Juju debug log: 
+
+<!-- (Optional) Copy the logs that are relevant to the bug & paste inside triple backticks below -->
```
Comment (reneradoi) [inline .github/workflows/ISSUE_TEMPLATE/bug_report.md]: Log output from `systemctl` would be more helpful here I guess.

**Review by reneradoi — DISMISSED** [review]
Fine for me, just adjust the bug template to a snap instead of a charm, then it can be merged.

File: .github/workflows/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (skourta) [reply .github/workflows/ISSUE_TEMPLATE/bug_report.md]: That is correct. I will copy the issue template from opensearch

**Review by reneradoi — DISMISSED** [review]

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          merge-multiple: true
+      
+      - name: Install snap package(s)
+        run: |
+          sudo snap install --dangerous charmed-etcd*.snap
+      
+      - name: Check etcd service status and run etcdctl and etcdutl
+        run: |
+          sudo systemctl status snap.charmed-etcd.daemon.service | grep 'active (running)'
+          charmed-etcd.etcd --version
```
Comment (Mehdi-Bendriss) [inline .github/workflows/ci.yaml]: we can do something more accurate here, i.e: 
```
expected_version="$(cat snap/snapcraft.yaml | yq .version)"

version="$(charmed-etcd.etcd --version | yq '.["etcd Version"]')"
if [[ "${version}" != "${expected_version}" ]]; then
    echo "Error: etcd Version is ${version}, expected {expected_version}"
    exit 1
fi
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      
+      - name: Run tests
+        run: |
+          charmed-etcd.etcdctl endpoint health | grep 'is healthy'
+          charmed-etcd.etcdctl put greeting "Hello, etcd" | grep 'OK'
+          charmed-etcd.etcdctl get greeting | grep 'Hello, etcd'
+      
+      - name: Run Performance checks
+        run: |
+          charmed-etcd.etcdctl check perf
```
Comment (Mehdi-Bendriss) [inline .github/workflows/ci.yaml]: What is the output / outcome of this check?

File: snap/snapcraft.yaml
Comment (Mehdi-Bendriss) [inline snap/snapcraft.yaml:1]: should be deleted.

File: snap/hooks/install [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh -e
+
+# This hook is called when the snap is installed
+# copy the configuration file to the snap's data directory
+cp $SNAP/config/etcd.conf.yml $SNAP_DATA/etcd.conf.yml
```
Comment (Mehdi-Bendriss) [inline snap/hooks/install]: nit:
```suggestion
cp $SNAP/config/etcd.conf.yml $SNAP_DATA/etcd.conf.yml

```

**Review by Mehdi-Bendriss — COMMENTED** [review]
Thanks Smail, I have a couple of small comments.

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (skourta) [reply .github/workflows/ci.yaml]: This runs a series of write/read and returns the throughput and the slowest request time.

**Review by Mehdi-Bendriss — APPROVED** [review]

**Review by reneradoi — APPROVED** [review]

**Pull request merged**